### PR TITLE
Ternary#84

### PR DESCRIPTION
**Pull request by jay51**
Ternary operator in variable definition and reassignment.

### CODE REVIEW

**Comment by jay51** [comment]
opps I didn't see your pull request to implement the ternary operator.

**Comment by jay51** [comment]
oh and you closed the issue. ok well this is what I was working on to implement the ternary operator,
you can discard it if the ternary operator is already working.

File: src/hermes_parser.c [hunk 1]
```diff
@@ -962,6 +972,15 @@ AST_T* hermes_parser_parse_function_definition(hermes_parser_T* hermes_parser, h
 
             ast_variable_definition->variable_value = hermes_parser_parse_expr(hermes_parser, scope);
 
+            if (strcmp(hermes_parser->current_token->value, "?") == 0){
```
Comment (sebbekarlsson) [inline src/hermes_parser.c:975]: Its better to compare integers than strings (faster)
like:

```C
if (hermes_parser->current_token->type == TOKEN_TERNARY)
{
    ...
}
```

File: src/hermes_parser.c [hunk 2]
```diff
@@ -319,10 +319,20 @@ AST_T* hermes_parser_parse_variable(hermes_parser_T* hermes_parser, hermes_scope
     {
         hermes_parser_eat(hermes_parser, TOKEN_EQUALS);
         AST_T* ast_assign = init_ast_with_line(AST_VARIABLE_ASSIGNMENT, hermes_parser->hermes_lexer->line_n);
+
         ast_assign->variable_assignment_left = ast_variable;
         ast_assign->variable_value = hermes_parser_parse_expr(hermes_parser, scope);
         ast_assign->scope = (struct hermes_scope_T*) scope;
 
+
+        if (strcmp(hermes_parser->current_token->value, "?") == 0){
```
Comment (sebbekarlsson) [inline src/hermes_parser.c:328]: Its better to compare integers than strings (faster)
like:

```C
if (hermes_parser->current_token->type == TOKEN_TERNARY)
{
    ...
}
```

**Comment by sebbekarlsson** [comment]
Great job!
I will unfortunately not merge this though, since ternary is already implemented.
How ever, I made some comments just for future pull requests.

Great job, I really appreciate it!

**Pull request closed without merging**